src/tainting/Taint.mli

```diff
@@ -25,29 +25,35 @@ type 'spec call_trace =
         * was used to produce the match, e.g. one of the `pattern-sources`.  *)
   | Call of AST_generic.expr * tainted_tokens * 'spec call_trace
       (** An indirect match through a function call. *)
-[@@deriving show]
 
-type arg_pos = { name : string; index : int } [@@deriving show]
+type arg = { name : string; index : int } [@@deriving show]
 (** A formal argument of a function given by its name and it's index/position. *)
 
-type arg_base =
+type base =
   | BGlob of IL.name  (** A global variable or a static class field. *)
   | BThis
       (** The 'this' or 'self' object, treated here like an special argument. *)
-  | BArg of arg_pos  (** A formal parameter in a function/method definition. *)
+  | BArg of arg  (** A formal parameter in a function/method definition. *)
```

Comment:
I think it would be nice to have some examples to illustrate those types. Is the goal here to represent
with a type the tainting "signature/summary" of a function?
Then maybe give a very simple example of a python/js/C/whatever function, and illustrate
on this function what will be the tainting signature with concrete example with BArg, BGlob, etc.

And then give another example where we see the need for a shape signature.